src/lib.rs

```diff
@@ -67,73 +76,118 @@ where
     {
         let idx = self.menu_idx;
-        if let Err(e) = self.window.add_menu_entry(idx, item_name) {
+        if let Err(e) = self.window.try_borrow_mut()?.add_menu_entry(idx, item_name) {
             return Err(e);
         }
         self.callback.insert(idx, make_callback(f));
         self.menu_idx += 1;
         Ok(idx)
     }
 
+    #[cfg(target_os = "macos")]
+    pub fn add_menu_item<F, E>(&mut self, item_name: &str, f: F) -> Result<u32, Error>
+    where
+        F: FnMut(&mut Application) -> Result<(), E> + Send + Sync + 'static,
+        E: error::Error + Send + Sync + 'static,
+    {
+        let idx = self.menu_idx;
+        if let Err(e) = self.window.try_borrow_mut()?.add_menu_entry(idx, item_name, make_callback(f)) {
+            return Err(e);
+        }
+        self.menu_idx += 1;
+        Ok(idx)
+    }
+
     pub fn add_menu_separator(&mut self) -> Result<u32, Error> {
         let idx = self.menu_idx;
-        if let Err(e) = self.window.add_menu_separator(idx) {
+        if let Err(e) = self.window.try_borrow_mut()?.add_menu_separator(idx) {
             return Err(e);
         }
         self.menu_idx += 1;
         Ok(idx)
     }
 
-    pub fn set_icon_from_file(&self, file: &str) -> Result<(), Error> {
-        self.window.set_icon_from_file(file)
+    pub fn set_icon_from_file(&mut self, file: &str) -> Result<(), Error> {
+        self.window.try_borrow_mut()?.set_icon_from_file(file)
     }
 
-    pub fn set_icon_from_resource(&self, resource: &str) -> Result<(), Error> {
-        self.window.set_icon_from_resource(resource)
+    pub fn set_icon_from_resource(&mut self, resource: &str) -> Result<(), Error> {
+        self.window.try_borrow_mut()?.set_icon_from_resource(resource)
     }
 
-    #[cfg(target_os = "windows")]
+    #[cfg(any(target_os = "windows", target_os = "macos"))]
     pub fn set_icon_from_buffer(
-        &self,
-        buffer: &[u8],
+        &mut self,
+        buffer: &'static [u8],
         width: u32,
         height: u32,
     ) -> Result<(), Error> {
-        self.window.set_icon_from_buffer(buffer, width, height)
+        self.window.try_borrow_mut()?.set_icon_from_buffer(buffer, width, height)
     }
 
-    pub fn shutdown(&self) -> Result<(), Error> {
-        self.window.shutdown()
+    pub fn shutdown(&mut self) -> Result<(), Error> {
+        self.window.try_borrow_mut()?.shutdown()
     }
 
-    pub fn set_tooltip(&self, tooltip: &str) -> Result<(), Error> {
-        self.window.set_tooltip(tooltip)
+    pub fn set_tooltip(&mut self, tooltip: &str) -> Result<(), Error> {
+        self.window.try_borrow_mut()?.set_tooltip(tooltip)
     }
 
-    pub fn quit(&mut self) {
-        self.window.quit()
+    pub fn quit(&mut self) -> Result<(), Error> {
+        self.window.try_borrow_mut()?.quit();
+        Ok(())
     }
 
-    pub fn wait_for_message(&mut self) -> Result<(), Error> {
+    #[cfg(not(target_os = "macos"))]
+    fn wait_for_message(&mut self) -> Result<(), Error> {
```

Comment:
You missed the visibility specifier here, meaning the example won't build on linux or windows. Just need to add pub (and I really need to get CI going for all platforms).

Reply:
Fixed in a5cc6bc